File: problems/2014/02_soccer/soccer.c


/*
   UCF 2014 (Fall) Local Programming Contest
   Problem: soccer
*/

#include <stdio.h>
#include <stdlib.h>

/* ************************************************************ */

int main(void)
{
   int   data_set_count, k, games, points, win, tie;
   FILE  *in_fptr;

   if ( (in_fptr = fopen("soccer.in", "r")) == NULL )
     {
      printf("*** can't open input file *** \n");
      exit(-1);
     }

   fscanf(in_fptr, "%d", &data_set_count);
   for ( k = 1;  k <= data_set_count;  ++k )
     {
      fscanf(in_fptr, "%d %d", &games, &points);
      printf("Team #%d\nGames: %d\nPoints: %d\nPossible records:\n",
                                                       k, games, points);

      for ( win = (points / 3);  win >= 0;  --win )
        {
         tie = points - (3 * win);
         if ( games >= (win + tie) )
            printf("%d-%d-%d\n", win, tie, (games - win - tie) );
        }

      printf("\n");

     }/* end for ( k ) */

   if ( fclose(in_fptr) == EOF )
     {
      printf("*** can't close input file *** \n");
      exit(-1);
     }

   return(0);

}/* end main */

/* ************************************************************ */

File: problems/2014/alternate/soccer.java

import java.util.*;

public class soccer
{
   public static void main(String[] args)
   {
      new soccer(new Scanner(System.in));
   }

   public soccer(Scanner in)
   {
      int T = in.nextInt();
      for (int tc=1; tc<=T; tc++)
      {
         int nGames = in.nextInt();
         int nPoints = in.nextInt();
         System.out.printf("Team #%d%n", tc);
         System.out.printf("Games: %d%n", nGames);
         System.out.printf("Points: %d%n", nPoints);
         System.out.printf("Possible records:%n");
         for (int numWins=nGames; numWins>=0; numWins--)
         {
            int points = numWins*3;
            int numTies = nPoints-points;
            int gamesLeft = nGames-numWins;
            if (numTies >= 0 && gamesLeft >= numTies)
               System.out.printf("%d-%d-%d%n", numWins, numTies, gamesLeft-numTies);
         }
         System.out.println();
      }
   }
}
